lib/compilers/clang.js

```diff
@@ -38,4 +45,32 @@ export class ClangCompiler extends BaseCompiler {
 
         return super.runCompiler(compiler, options, inputFilename, execOptions);
     }
+
+    async extractDeviceCode(result, filters, outputFilename) {
+        // Check to see if there is any offload code in the assembly file.
+        const offload_regexp = /__CLANG_OFFLOAD_BUNDLE__/;
+        if (!offload_regexp.exec(result.asm) || !this.compiler.deviceExtractor)
+            return result;
+
+        const extractor = this.compiler.deviceExtractor;
+        const extractor_result = await this.exec(extractor,
+            [outputFilename, result.dirPath + "/devices"], {});
+        if (extractor_result.code != 0) {
+            // XXX: What do I do now?
+            throw new Error(`Extractor failed: ${extractor_result.stderr}`);
```

Comment:
Generally we wouldn't throw in this case, you would modify `result` to contain the error information in some way and then return early. This way the error information flows to the user on the client side.